hts-precompile/TokenManagementContract.sol

```diff
@@ -0,0 +1,416 @@
+    function isKycPublic(address token, address account)external returns (int64 responseCode, bool kycGranted) {
+        (responseCode, kycGranted) = this.isKyc(token, account);
+
+        if (responseCode != HederaResponseCodes.SUCCESS) {
+            revert();
+        }
+    }
+
+    function grantTokenKycPublic(address token, address account)external returns (int64 responseCode) {
+        (responseCode) = this.grantTokenKyc(token, account);
+
+        if (responseCode != HederaResponseCodes.SUCCESS) {
+            revert();
+        }
+    }
+
+    function revokeTokenKycPublic(address token, address account)external returns (int64 responseCode) {
+        (responseCode) = this.revokeTokenKyc(token, account);
+
+        if (responseCode != HederaResponseCodes.SUCCESS) {
+            revert();
+        }
+    }
+
+    function getTokenExpiryInfoPublic(address token)external returns (int responseCode, IHederaTokenService.Expiry memory expiryInfo) {
+        (responseCode, expiryInfo) = this.getTokenExpiryInfo(token);
+
+        if (responseCode != HederaResponseCodes.SUCCESS) {
+            revert();
+        }
+    }
+
+    function updateTokenExpiryInfoPublic(address token, IHederaTokenService.Expiry memory expiryInfo)external returns (int responseCode) {
+        (responseCode) = this.updateTokenExpiryInfo(token, expiryInfo);
+
+        if (responseCode != HederaResponseCodes.SUCCESS) {
+            revert();
+        }
+    }
+
+    function isTokenPublic(address token) public returns (int64 responseCode, bool isTokenFlag) {
+        (responseCode, isTokenFlag) = HederaTokenService.isToken(token);
+
+        if (responseCode != HederaResponseCodes.SUCCESS) {
+            revert();
+        }
+    }
+
+    function getTokenTypePublic(address token) public returns (int64 responseCode, int32 tokenType) {
+        (responseCode, tokenType) = HederaTokenService.getTokenType(token);
+
+        if (responseCode != HederaResponseCodes.SUCCESS) {
+            revert();
+        }
+    }
+
+    function getTokenKeyPublic(address token, uint keyType) external returns (int64 responseCode, IHederaTokenService.KeyValue memory key) {
+        (responseCode, key) = HederaTokenService.getTokenKey(token, keyType);
+
+        if (responseCode != HederaResponseCodes.SUCCESS) {
+            revert();
+        }
+    }
+
+    function updateTokenInfoPublic(address token, IHederaTokenService.HederaToken memory tokenInfo) public returns (int responseCode) {
+        (responseCode, tokenInfo) = HederaTokenService.updateTokenInfo(token, tokenInfo);
+
+        if (responseCode != HederaResponseCodes.SUCCESS) {
+            revert();
+        }
+    }
+
+    function updateTokenKeysPublic(address token, IHederaTokenService.TokenKey[] memory keys) public returns (int64 responseCode) {
+        (responseCode, keys) = HederaTokenService.updateTokenKeys(token, keys);
+
+        if (responseCode != HederaResponseCodes.SUCCESS) {
+            revert();
+        }
+    }
```

Comment:
These are all just renamed copies of functions we have in [SafeHTS.sol](https://github.com/hashgraph/hedera-smart-contracts/blob/main/safe-hts-precompile/SafeHTS.sol) - why do we need another set?